Advanced-Audio-Driver-Java/app/src/main/java/com/tokbox/sample/advancedaudiodriver/AdvancedAudioDevice.java

```diff
@@ -1000,13 +1002,32 @@ private void unregisterBtReceiver() {
 
     private boolean isPhoneStateListenerRegistered;
 
+    //Phone state permissions are required to from Api 31. Adding this check to avoid crash.
+    private boolean hasPhoneStatePermission() {
+        if (Build.VERSION.SDK_INT >= 31) {
```

Comment:
Why two `if` statements? I always find it harder to read than `&&`. That said, I am not asking to change it :)

Reply:
Yes, I know @jintgeorge likes it. 👍 Just wondering if I am in the minority who finds it harder to read. 🤷🏼‍♂️ No need to change anyway.

Reply:
lol. Don't know why I did like that. Feel free to change it. We were working to solve the crash issue and I wasn't worried much about the code.